File: push/src/main/java/com/xiaomi/mipush/sdk/stat/db/MessageInsertJob.java
package com.xiaomi.mipush.sdk.stat.db;

import android.content.ContentValues;
import android.content.Context;
import com.xiaomi.mipush.sdk.stat.PushStatClientManager;
import com.xiaomi.mipush.sdk.stat.db.MessageInfoContract;
import com.xiaomi.mipush.sdk.stat.db.base.DbManager;
import com.xiaomi.xmpush.thrift.ClientUploadDataItem;
import com.xiaomi.xmpush.thrift.XmPushThriftSerializeUtils;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/stat/db/MessageInsertJob.class */
public class MessageInsertJob extends DbManager.InsertJob {
    private String mDescription;

    public MessageInsertJob(String str, ContentValues contentValues, String str2) {
        super(str, contentValues);
        this.mDescription = "MessageInsertJob";
        this.mDescription = str2;
    }

    public static MessageInsertJob buildInsertJob(Context context, String str, ClientUploadDataItem clientUploadDataItem) {
        byte[] bArrConvertThriftObjectToBytes = XmPushThriftSerializeUtils.convertThriftObjectToBytes(clientUploadDataItem);
        if (bArrConvertThriftObjectToBytes == null || bArrConvertThriftObjectToBytes.length <= 0) {
            return null;
        }
        ContentValues contentValues = new ContentValues();
        contentValues.put("status", (Integer) 0);
        contentValues.put("messageId", "");
        contentValues.put(MessageInfoContract.MessageEntry.COLUMN_NAME_MESSAGE_ITEM_ID, clientUploadDataItem.getId());
        contentValues.put(MessageInfoContract.MessageEntry.COLUMN_NAME_MESSAGE_ITEM, bArrConvertThriftObjectToBytes);
        contentValues.put(MessageInfoContract.MessageEntry.COLUMN_NAME_APP_ID, PushStatClientManager.getInstance(context).getAppId());
        contentValues.put(MessageInfoContract.MessageEntry.COLUMN_NAME_APP_PACKAGE_NAME, PushStatClientManager.getInstance(context).getPackageName());
        contentValues.put(MessageInfoContract.MessageEntry.COLUMN_NAME_CREATE_TIMESTAMP, Long.valueOf(System.currentTimeMillis()));
        contentValues.put(MessageInfoContract.MessageEntry.COLUMN_NAME_UPLOAD_TIMESTAMP, (Integer) 0);
        return new MessageInsertJob(str, contentValues, "a job build to insert message to db");
    }

    @Override // com.xiaomi.mipush.sdk.stat.db.base.DbManager.InsertJob, com.xiaomi.mipush.sdk.stat.db.base.DbManager.BaseJob
    public String description() {
        return this.mDescription + "";
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/mipush/sdk/stat/db/MessageInsertJob.kt
package com.xiaomi.mipush.sdk.stat.db

import android.content.ContentValues
import android.content.Context
import com.xiaomi.mipush.sdk.stat.db.base.DbManager
import com.xiaomi.mipush.sdk.stat.PushStatClientManager
import com.xiaomi.xmpush.thrift.ClientUploadDataItem
import com.xiaomi.xmpush.thrift.XmPushThriftSerializeUtils

open class MessageInsertJob(
    str: String,
    contentValues: ContentValues,
    var mDescription: String,
) : DbManager.InsertJob(str, contentValues) {

    companion object {
        fun buildInsertJob(
            context: Context,
            str: String,
            clientUploadDataItem: ClientUploadDataItem,
        ): MessageInsertJob? {
            val bArrConvertThriftObjectToBytes =
                XmPushThriftSerializeUtils.convertThriftObjectToBytes(clientUploadDataItem)
            if (bArrConvertThriftObjectToBytes == null || bArrConvertThriftObjectToBytes.isEmpty()) {
                return null
            }
            val contentValues = ContentValues()
            contentValues.put("status", 0)
            contentValues.put("messageId", "")
            contentValues.put(
                MessageInfoContract.MessageEntry.COLUMN_NAME_MESSAGE_ITEM_ID,
                clientUploadDataItem.id,
            )
            contentValues.put(
                MessageInfoContract.MessageEntry.COLUMN_NAME_MESSAGE_ITEM,
                bArrConvertThriftObjectToBytes,
            )
            contentValues.put(
                MessageInfoContract.MessageEntry.COLUMN_NAME_APP_ID,
                PushStatClientManager.getInstance(context).appId,
            )
            contentValues.put(
                MessageInfoContract.MessageEntry.COLUMN_NAME_APP_PACKAGE_NAME,
                PushStatClientManager.getInstance(context).packageName,
            )
            contentValues.put(
                MessageInfoContract.MessageEntry.COLUMN_NAME_CREATE_TIMESTAMP,
                System.currentTimeMillis(),
            )
            contentValues.put(MessageInfoContract.MessageEntry.COLUMN_NAME_UPLOAD_TIMESTAMP, 0)
            return MessageInsertJob(
                str,
                contentValues,
                "a job build to insert message to db",
            )
        }
    }

    override fun description(): String = mDescription
}
